Elasticsearch: failed to start

Error start elasticsearch service:
```
сен 15 01:13:12 tech systemd[1]: Starting Elasticsearch Daemon...
сен 15 01:13:17 tech elasticsearch[43247]: [2021-09-15T01:13:17,399][WARN ][o.e.c.l.LogConfigurator  ] [unknown] Some logging configurations have %marker but don't have %node_name. We will automatically add %node_name to the pattern to ease the migration for users who customize log4j2.properties but will stop this behavior in 7.0. You should manually replace `%node_name` with `[%node_name]%marker ` in these locations:
сен 15 01:13:17 tech elasticsearch[43247]:   /var/lib/elasticsearch/config/log4j2.properties
сен 15 01:13:18 tech elasticsearch[43247]: [2021-09-15T01:13:18,423][INFO ][o.e.e.NodeEnvironment    ] [ycjzZOJ] using [1] data paths, mounts [[/ (zvm11-tech/nixos/root1)]], net usable_space [17.4gb], net total_space [18.2gb], types [zfs]
сен 15 01:13:18 tech elasticsearch[43247]: [2021-09-15T01:13:18,424][INFO ][o.e.e.NodeEnvironment    ] [ycjzZOJ] heap size [990.7mb], compressed ordinary object pointers [true]
сен 15 01:13:18 tech elasticsearch[43247]: [2021-09-15T01:13:18,428][INFO ][o.e.n.Node               ] [ycjzZOJ] node name derived from node ID [ycjzZOJKSv26SV6CQJya1g]; set [node.name] to override
сен 15 01:13:18 tech elasticsearch[43247]: [2021-09-15T01:13:18,429][INFO ][o.e.n.Node               ] [ycjzZOJ] version[6.8.3], pid[43247], build[default/tar/0c48c0e/2019-08-29T19:05:24.312154Z], OS[Linux/5.12.19-hardened1/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/1.8.0_272/25.272-b10]
сен 15 01:13:18 tech elasticsearch[43247]: [2021-09-15T01:13:18,429][INFO ][o.e.n.Node               ] [ycjzZOJ] JVM arguments [-Xms1g, -Xmx1g, -XX:+UseConcMarkSweepGC, -XX:CMSInitiatingOccupancyFraction=75, -XX:+UseCMSInitiatingOccupancyOnly, -Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.io.tmpdir=/tmp/elasticsearch-5208053257500823123, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=data, -XX:ErrorFile=logs/hs_err_pid%p.log, -XX:+PrintGCDetails, -XX:+PrintGCDateStamps, -XX:+PrintTenuringDistribution, -XX:+PrintGCApplicationStoppedTime, -Xloggc:/var/lib/elasticsearch/logs/gc.log, -XX:+UseGCLogFileRotation, -XX:NumberOfGCLogFiles=32, -XX:GCLogFileSize=64m, -Des.path.home=/var/lib/elasticsearch, -Des.path.conf=/var/lib/elasticsearch/config, -Des.distribution.flavor=default, -Des.distribution.type=tar]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,545][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [aggs-matrix-stats]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,546][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [analysis-common]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,546][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [ingest-common]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,547][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [ingest-geoip]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,547][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [ingest-user-agent]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,547][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [lang-expression]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,547][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [lang-mustache]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,547][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [lang-painless]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,548][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [mapper-extras]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,548][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [parent-join]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,548][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [percolator]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,548][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [rank-eval]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,548][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [reindex]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,549][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [repository-url]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,549][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [transport-netty4]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,549][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [tribe]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,550][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-ccr]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,550][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-core]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,550][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-deprecation]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,550][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-graph]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,550][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-ilm]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,551][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-logstash]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,551][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-ml]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,551][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-monitoring]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,551][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-rollup]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,552][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-security]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,552][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-sql]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,552][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-upgrade]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,552][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] loaded module [x-pack-watcher]
сен 15 01:13:23 tech elasticsearch[43247]: [2021-09-15T01:13:23,553][INFO ][o.e.p.PluginsService     ] [ycjzZOJ] no plugins loaded
сен 15 01:13:27 tech elasticsearch[43247]: [2021-09-15T01:13:27,592][WARN ][o.e.b.ElasticsearchUncaughtExceptionHandler] [ycjzZOJ] uncaught exception in thread [main]
сен 15 01:13:27 tech elasticsearch[43247]: org.elasticsearch.bootstrap.StartupException: java.lang.IllegalArgumentException: unknown setting [gateway.auto_import_dangling_indices] please check that any required plugins are installed, or check the breaking changes documentation for removed settings
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:163) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:150) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124) ~[elasticsearch-cli-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.cli.Command.main(Command.java:90) ~[elasticsearch-cli-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:116) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:93) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]: Caused by: java.lang.IllegalArgumentException: unknown setting [gateway.auto_import_dangling_indices] please check that any required plugins are installed, or check the breaking changes documentation for removed settings
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:530) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:475) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:446) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:417) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.common.settings.SettingsModule.<init>(SettingsModule.java:148) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.node.Node.<init>(Node.java:374) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.node.Node.<init>(Node.java:266) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:212) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:212) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:333) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:159) ~[elasticsearch-6.8.3.jar:6.8.3]
сен 15 01:13:27 tech elasticsearch[43247]:         ... 6 more
сен 15 01:13:27 tech systemd[1]: elasticsearch.service: Main process exited, code=exited, status=1/FAILURE
сен 15 01:14:42 tech systemd[1]: elasticsearch.service: start-post operation timed out. Terminating.
сен 15 01:14:42 tech systemd[1]: elasticsearch.service: Control process exited, code=killed, status=15/TERM
сен 15 01:14:42 tech systemd[1]: elasticsearch.service: Failed with result 'exit-code'.
сен 15 01:14:42 tech systemd[1]: Failed to start Elasticsearch Daemon.
сен 15 01:14:42 tech systemd[1]: elasticsearch.service: Consumed 23.240s CPU time, received 8.5K IP traffic, sent 10.6K IP traffic.

```
### Steps To Reproduce
Configuration:
```
  services.elasticsearch = {
    enable = true;
    package = pkgs.elasticsearch;
    listenAddress = "0.0.0.0";
    port = 9200;
  };
```
### Expected behavior
Used master branch

### Metadata
 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.19-hardened1, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos-21.11pre306131.39bce8345f3"`
 - channels(user): `"nixpkgs-2009-20.09beta1346.05334ad7852"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

```

--- a/nixos/modules/services/search/elasticsearch.nix
+++ b/nixos/modules/services/search/elasticsearch.nix
@@ -5,13 +5,13 @@ with lib;
 let
   cfg = config.services.elasticsearch;
 
+  es7 = builtins.compareVersions cfg.package.version "7" >= 0;
+
   esConfig = ''
     network.host: ${cfg.listenAddress}
     cluster.name: ${cfg.cluster_name}
-    ${lib.optionalString cfg.single_node ''
-      discovery.type: single-node
-      gateway.auto_import_dangling_indices: true
-    ''}
+    ${lib.optionalString cfg.single_node "discovery.type: single-node"}
+    ${lib.optionalString (cfg.single_node && es7) "gateway.auto_import_dangling_indices: true"}
 
     http.port: ${toString cfg.port}
     transport.port: ${toString cfg.tcp_port}

